documentstore/domain.py

```diff
@@ -592,6 +592,13 @@ def insert_item(
         items_bundle: dict, index: int, item: str, now: Callable[[], str] = utcnow
     ) -> dict:
         if item in items_bundle["items"]:
+            if items_bundle["items"].index(item) != index:
```

Comment:
Não acredito que isso se tornará um problema, mas esse código potencialmente itera sobre praticamente todos os itens de `items_bundle["items"]` 3 vezes.  Em Python as operações `list.index` e `in` possuem complexidade O(n).

Reply:
O que você pensa sobre ao invés de termos 2 possibilidades distintas de `AlreadyExists` sendo retornada por esse método (setando `position` ou não), retornarmos sempre com o valor definido? Assim, a view function ao invés de checar se o valor é diferente de `None`, checaria se o valor é o mesmo de `index`.

Reply:
Acho que simplifica no caso de uso que nós temos hoje mas caso queiramos que o `patch` aceite uma lista de issues com ou sem índices esta implementação vai tornar inviável sabermos qual issue levantou a exceção. Talvez isso seja uma indicio de que nós não devamos aceitar esse tipo payload.